code/controllers/CMSMain.php

```diff
@@ -88,8 +97,24 @@ public function init() {
 
 		CMSBatchActionHandler::register('publish', 'CMSBatchAction_Publish');
 		CMSBatchActionHandler::register('unpublish', 'CMSBatchAction_Unpublish');
-		CMSBatchActionHandler::register('delete', 'CMSBatchAction_Delete');
-		CMSBatchActionHandler::register('deletefromlive', 'CMSBatchAction_DeleteFromLive');
+		
+
+		// Check legacy actions
+		$legacy = $this->config()->enabled_legacy_actions;
```

Comment:
The problem with the aforementioned config option is that it doesn't constrain itself only to apply to CMSMain, so this is needed.